tests/CloudFront/SignerTest.php

```diff
@@ -25,6 +25,21 @@ public function setUp()
         );
     }
 
+    /**
+     * Assert that the key is parsed during construction
+     *
+     * @expectedException \InvalidArgumentException
+     * @expectedExceptionMessageRegExp /PEM .*no start line/
+     */
+    public function testBadPrivateKey() {
+        $filename = tempnam(sys_get_temp_dir(), 'cloudfront-fake-key');
+        file_put_contents($filename, "Not a real private key");
```

Comment:
Can you clean this up with an `unlink` after use? You may need to catch and rethrow the exception.

Reply:
Done!  Went with a `finally` cause it's available in php 5.5 and it looks like this repo targets php 5.5 and above.